Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -217,6 +211,121 @@ void AODProducerWorkflowDPL::fillTracksTable(const TTracks& tracks, std::vector<
   }
 }
 
+template <typename TMCTruthITS, typename TMCTruthTPC>
+void AODProducerWorkflowDPL::findRelatives(const o2::steer::MCKinematicsReader& mcReader, const TMCTruthITS& mcTruthITS, const TMCTruthTPC& mcTruthTPC,
+                                           std::vector<std::vector<std::vector<int>>>& toStore)
+{
+  if (mFillTracksITS) {
+    for (auto& mcTruth : mcTruthITS) {
+      int source = mcTruth.getSourceID();
+      int event = mcTruth.getEventID();
+      int track = mcTruth.getTrackID();
+      toStore[source][event][track] = 1;
+    }
+  }
+  if (mFillTracksTPC) {
+    for (auto& mcTruth : mcTruthTPC) {
+      int source = mcTruth.getSourceID();
+      int event = mcTruth.getEventID();
+      int track = mcTruth.getTrackID();
+      toStore[source][event][track] = 1;
+    }
+  }
+  for (int source = 0; source < mcReader.getNSources(); source++) {
+    for (int event = 0; event < mcReader.getNEvents(source); event++) {
+      std::vector<MCTrack> const& mcParticles = mcReader.getTracks(source, event);
```

Comment:
there are 2 places that fetch the tracks. Are both of them needed?

Reply:
Hi @sawenzel. I updated the code and merged `findRelatives()` and `fillMCParticlesTable()` into one function to fetch MC tracks only once (see latest commit)